### PR TITLE
feature: adding stg_patiens model and corresponding yaml with the doc…

### PR DESCRIPTION
**Pull request by cojovero**
feature: adding stg_patients model and corresponding yaml with the documentation

## Screenshots:

Screenshot 2023-06-07 at 3 05 05 PM
## Checklist:

- [x] My pull request represents one logical piece of work.
- [x] My commits are related to the pull request and look clean.
- [x] My SQL follows the [dbt Labs style guide](https://github.com/dbt-labs/corp/blob/master/dbt_style_guide.md).
- [x] I have materialized my models appropriately.
- [x] I have added appropriate tests and documentation to any new models.
- [ ] I have updated the README file.
- [ ] I have mask the PII data.

### CODE REVIEW

File: models/marts/staging/stg_patients.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    select * from {{ source('demo_healthcare', 'patients_view') }}
+
+),
+
+final as (
+
+    select 
+        -- primary key
+        id as patient_id,
+        gender as patient_gender,
```
Comment (90awlui) [inline models/marts/staging/stg_patients.sql:13]: can you separate as well into 

```
-- timestamps 
...

-- fields 
...
```

File: models/marts/staging/stg_patients.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        birthDate as patient_birth_date,
+        name[SAFE_OFFSET(0)].family as patient_last_name,
+        name[SAFE_OFFSET(0)].given as patient_first_name,
+        name[SAFE_OFFSET(0)].use as name_in_use,
+        deceased.dateTime as deceased_at,
```
Comment (90awlui) [inline models/marts/staging/stg_patients.sql]: do these also need a `patient_` prefix?

File: models/marts/staging/stg_patients.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      - name: patient_birth_date
+        description: "{{ doc('patient_birth_date')}}"
+
+      - name: patient_last_name
+        description: "{{ doc('patient_last_name')}}"
+        meta: 
+            masking_policy: full_encrypt_pii  
+        meta: 
+            masking_policy: full_encrypt_pii
```
Comment (90awlui) [inline models/marts/staging/stg_patients.yml]: duplicated

File: models/marts/staging/stg_patients.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        description: "{{ doc('patient_id')}}"
+        tests:
+          - unique
+          - not_null
+
+      - name: patient_gender
+        description: "{{ doc('patient_gender')}}"
+
+      - name: patient_birth_date
+        description: "{{ doc('patient_birth_date')}}"
```
Comment (90awlui) [inline models/marts/staging/stg_patients.yml:17]: this should probably be masked

File: models/marts/staging/stg_patients.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      - name: patient_first_name
+        description: "{{ doc('patient_first_name')}}"
+        meta: 
+            masking_policy: full_encrypt_pii 
+            
+      - name: name_in_use
+        description: "{{ doc('name_in_use')}}"
+
+      - name: deceased_at
+        description: "{{ doc('deceased_at')}}"
```
Comment (90awlui) [inline models/marts/staging/stg_patients.yml:29]: do these need to be masked?

**Review by 90awlui — CHANGES_REQUESTED** [review]

File: models/marts/staging/stg_patients.yml (hunk 5 above)
Comment (cojovero) [reply models/marts/staging/stg_patients.yml:29]: We may need to resort to masking in Looker, so i've removed these for now.

**Review by 90awlui — APPROVED** [review]

**Pull request merged**